doc/02_user/configuration.rst

```diff
@@ -0,0 +1,43 @@
+.. _ibex-config:
+
+Ibex Configurations
+===================
+
+The ``ibex_top`` module has a large number of top-level parameters which configure the core (see :ref:`core-integration`).
+This gives rise to a huge number of possible Ibex core configurations.
+To manage this complexity a number of named configurations is provided in the :file:`ibex_configs.yml` file.
+A subset of these are 'supported configurations' which are the focus of verification and development activities.
+
+Configuration Tool
+------------------
+
+A tool :file:`util/ibex_config.py` is provided to work with the named configurations.
+This tool provides command line options to set Ibex parameters for various EDA tools for a named configuration.
+Various Ibex flows (e.g. the DV flow) use this tool internally and can be provided with a configuration name from :file:`util/ibex_config.py` to work with.
+
+Here is an example of using the configuration tool to get the FuseSoC options required to build the ``opentitan`` configuration.
+
+.. code-block:: bash
+
+  # Request FuseSoC options required to build the 'opentitan' Ibex configuration.
+  ./util/ibex_config.py opentitan fusesoc_opts
+
+  # The output of the tool
+  --RV32E=0 --RV32M=ibex_pkg::RV32MSingleCycle --RV32B=ibex_pkg::RV32BOTEarlGrey --RegFile=ibex_pkg::RegFileFF --BranchTargetALU=1 --WritebackStage=1 --ICache=1 --ICacheECC=1 --ICacheScramble=1 --BranchPredictor=0 --DbgTriggerEn=1 --SecureIbex=1 --PMPEnable=1 --PMPGranularity=0 --PMPNumRegions=16 --MHPMCounterNum=10 --MHPMCounterWidth=32
+
+For further information about using the tool check the help provided on the command line.
+
+.. code-block:: bash
+
+  # Get help on using ibex_config.py
+  ./util/ibex_config.py -h
+
+Supported Configurations
+------------------------
+
+The current set of supported configurations are:
```

Comment:
What about `maxperf-pmp-bmfull` and `maxperf-pmp-bmfull-icache`?

Reply:
We want to keep the set of supported configs low. We've got testing of the icache in 'opentitan' (as well as the standalone icache testbench). We lack testing for the 'full' bitmanip config, though it is potentially an odd config as it involves a mix of ratified and un-ratified spec (so does 'opentitan' but that it motivated by particular opentitan requirements).

So perhaps we should add a bmfull config at some point, but for now I prefer to just keep things as is. We do need to think about our set of supported bitmanip extensions and what makes sense.

Reply:
I'll keep maxperf-pmp-bmfull in CI, to ensure it doesn't break though will leave it out of supported configurations for now.